crates/autopilot/src/run_loop.rs

```diff
@@ -85,6 +85,7 @@ impl RunLoop {
         // Shuffle so that sorting randomly splits ties.
         solutions.shuffle(&mut rand::thread_rng());
         solutions.sort_unstable_by_key(|solution| solution.1.score);
+        let competition_simulation_block = self.current_block.borrow().number;
```

Comment:
This can be inaccurate.
I think at the moment our logic simply simulates using the `latest` block which may be different across these simulations. If we want to get an accurate reading we probably need to use some helper solidity code to return the used block number alongside the simulation result using state overrides.
That way we would know exactly which block number was used for each solution.

Reply:
At the moment we don't do any simulations in the colocated runloop. Maybe we could implement that when we add simulations (if we add them ever).
For now, this line just mimics what the non-colocated [driver is doing right now](https://github.com/cowprotocol/services/blob/main/crates/solver/src/driver.rs#L359-L361).

Reply:
Ahh, true. Do you know what script or so needs the simulation block number and whether this worse than before approximation is still good enough?
If we can get by without this block number maybe it's worth it. This might require changes in other software so I think it's fine to leave this in this PR.

Reply:
[Here](https://github.com/cowprotocol/solver-rewards/blob/b3783694e36f66b266ec34f7c05b0f3e1db0d7f7/internal_transfers/actions/src/accounting.ts#L66) it is.  To me, it seems like it will do the job.

Reply:
Ofc, will need to update the solver rewards script before colocation is enabled.